controllers/awscluster_controller.go

```diff
@@ -352,6 +372,12 @@ func (r *AWSClusterReconciler) reconcileNormal(clusterScope *scope.ClusterScope)
 	}
 	conditions.MarkTrue(awsCluster, infrav1.S3BucketReadyCondition)
 
+	if err := iamService.ReconcileOIDCProvider(context.TODO()); err != nil {
```

Comment:
Should we wrap this with a test of the feature being enabled to be explicit? I know the validating webook shouldn't allow setting the associate field when the feature flag is disabled. Undecided about this if i'm honest ;)

Reply:
Done